lib/apiservers/engine/backends/eventmonitor.go

```diff
@@ -236,8 +238,21 @@ func (p DockerEventPublisher) PublishEvent(event plevents.BaseEvent) {
 			attrs["exitCode"] = code
 			log.Infof("Sending die event for container %s - code: %s", vc.ContainerID, code)
 			actor := CreateContainerEventActorWithAttributes(vc, attrs)
-
 			EventService().Log(containerDieEvent, eventtypes.ContainerEventType, actor)
+
+			// auto-remove if required
+			if vc.HostConfig.AutoRemove {
+				config := &types.ContainerRmConfig{
+					ForceRemove:  true,
+					RemoveVolume: true,
```

Comment:
@matthewavery The definitive answer should be in the containerVM configs - if we chose to cache that, it's an optimization. First we need to decide what occurs if a volume is:
a. in use by a running container
b. referenced by a stopped container
c. what the behaviour of the stopped container should be if a referenced volume is deleted and the container is started, both via `docker start` and via vSphere